src/Renci.SshNet/Session.cs

```diff
@@ -1047,9 +1047,9 @@ internal void SendMessage(Message message)
                 WaitOnHandle(_keyExchangeCompletedWaitHandle.WaitHandle);
             }
 
-            if (_logger.IsEnabled(LogLevel.Debug))
+            if (_logger.IsEnabled(LogLevel.Trace))
             {
-                _logger.LogDebug("[{SessionId}] Sending message '{MessageType}' to server: '{Message}'.", SessionIdHex, message.GetType().Name, message);
+                _logger.LogTrace("[{SessionId}] Sending message {MessageName}({MessageNumber}) to server: '{Message}'.", SessionIdHex, message.MessageName, message.MessageNumber, message.ToString());
```

Comment:
I changed it to output e.g. "Sending message SSH_MSG_NEWKEYS(21)" as it is helpful to see the number as well. It is a bit duplicative atm because most messages include the message name in the `ToString`

also I added explicit `ToString` on the message in case e.g. a json logger would dump all the properties of the message. trying to strike a balance between useful information and private session information

Reply:
also, wondering if logger.BeginScope would have a valid use for tagging the SessionId to logs

Reply:
Using a scope for SessionId would make sense. A logging scope is based on an AsyncLocal though, so the BeginScope would have to be called in quite a few places.